recipes/libpqxx/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+
+        if self.options.shared and \
+           self.settings.os == "Windows" and \
+           compiler == "Visual Studio":
+            self.output.info("Override libpq:shared to True.")
+            self.options["libpq"].shared = True
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            args = [
+                "--disable-documentation",
+                "--with-postgres-include={}".format(os.path.join(self.deps_cpp_info["libpq"].rootpath, "include")),
+                "--with-postgres-lib={}".format(os.path.join(self.deps_cpp_info["libpq"].rootpath, "lib")),
+                "--enable-static={}".format("no" if self.options.shared else "yes"),
+                "--enable-shared={}".format("yes" if self.options.shared else "no")
+            ]
+            self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+            env_vars = self._autotools.vars
+            env_vars["PG_CONFIG"] = os.path.join(self.deps_cpp_info["libpq"].rootpath, "bin", "pg_config")
+            self._autotools.configure(args=args, vars=env_vars)
+        return self._autotools
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_DOC"] = False
+        cmake.definitions["BUILD_TEST"] = False
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def _patch_files(self):
+        if self.version in self.conan_data["patches"]:
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
+
+    def build(self):
+        self._patch_files()
+
+        if self._using_cmake:
+            cmake = self._configure_cmake()
+            cmake.build()
+        else:
+            with tools.chdir(self._source_subfolder):
+                autotools = self._configure_autotools()
+                autotools.make()
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        if self._using_cmake:
+            cmake = self._configure_cmake()
+            cmake.install()
+
+        else:
+            with tools.chdir(self._source_subfolder):
+                autotools = self._configure_autotools()
+                autotools.install()
+
+    def package_info(self):
+        pqxx_with_suffix = "pqxx-%s.%s" % tuple(self.version.split(".")[0:2])
+        is_package_with_suffix = not (self._using_cmake and self.settings.os == "Windows") and \
+                                 self.options.shared
+        self.cpp_info.libs.append(pqxx_with_suffix if is_package_with_suffix else "pqxx")
+
+        if self.settings.os == "Windows":
+            self.cpp_info.libs.extend(["wsock32 ", "Ws2_32"])
```

Comment:
these should be added to system_libs:
```suggestion
            self.cpp_info.system_libs.extend(["wsock32 ", "ws2_32"])
```